test/recipes/91-test_pkey_check.t

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env perl
+# Copyright 2017-2020 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use strict;
+use warnings;
+
+use File::Spec;
+use OpenSSL::Test qw/:DEFAULT data_file/;
+use OpenSSL::Test::Utils;
+
+sub check_key {
+    my $f = shift;
+
+    return run(app(['openssl', 'pkey', '-check', '-text',
+                    '-in', $f]));
+}
+
+sub check_key_notok {
+    my $f = shift;
+    my $str = "$f should fail validation";
+
+    $f = data_file($f);
+
+    if ( -s $f ) {
+        ok(!check_key($f), $str);
+    } else {
+        diag("Missing file $f");
+        ok(0, $str);
```

Comment:
@levitte , stylewise is there a better way to write this?
I saw https://perldoc.perl.org/Test::More#fail would it be more appropriate?

Reply:
One of these:

```
    if (-s $f) {
        ok(!check_key($f), $str);
    } else {
        fail("Missing file $f");
    }
```

or

``` perl
    SKIP: {
        skip "Missing file $f", 1 unless -s $f;
        ok(!check_key($f), $str);
    }
```

Which variant you choose depends on what you think this recipe is meant to test.  Is it meant to test the presence of those files?  In that case the former.  Is it only meant to test that `openssl pkey` fails for certain keys?  Then the latter seems more appropriate.

Reply:
This is tricky!!

I conceptually and stylistically like the second option with the `SKIP` block... but, what I don't like is that I would consider the absence (because it was removed or renamed) of one of the files explicitly mentioned in the test recipe as a severe error in the recipe, and that is why functionally I would prefer the first solution........

Can you help me decide among the two options? You also have a better perspective on what most other recipes do about similar cases, so I would go with the option you prefer here.

Reply:
The only similar test that I can think of that does the same kind of check is the fuzz test.  There, we have explicitly allowed the corpora to be absent.  From your description, this one is different and you do want to have a check that the files are indeed present, and since you otherwise check that the program invocation fails (which is the actual unusual bit), that check must be explicit in this recipe.

With all that in mind, the first variant (the one with the `fail` call) seems to be the most appropriate.

Reply:
Addressed with 05bd0080fc1cc59a10a6a7b9b1818c763ebec1a5